runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java

```diff
@@ -259,28 +278,44 @@ static StreamExecutionEnvironment createStreamExecutionEnvironment(
         options.setShutdownSourcesAfterIdleMs(0L);
       }
     }
+  }
 
-    applyLatencyTrackingInterval(flinkStreamEnv.getConfig(), options);
-
-    if (options.getAutoWatermarkInterval() != null) {
-      flinkStreamEnv.getConfig().setAutoWatermarkInterval(options.getAutoWatermarkInterval());
-    }
-
-    // State backend
-    if (options.getStateBackendFactory() != null) {
+  private static void configureStateBackend(
```

Comment:
Flink is headed in the direction where everything that is set on an environment becomes configurable (including the executor, FLIP-73). This change kind of goes in the opposite direction, increasing the amount of Flink pipeline options further. Should we look into the generic configuration mechanism instead, where it is really easy for the user to supply the Flink configuration (optionally inline, instead of via a separate file)?

Reply:
I think that's a great idea. We can start thinking about that on the mailing list and handle this via separate JIRA issue. This pipeline option can then be replaced by the generic configuration option.

For now, this pipeline option will fulfill a common request by Beam users to directly set the state backend without having to change the Flink configuration.

Reply:
Plus, we want to be able to easily run Nexmark with RocksDB.

Reply:
Is the intention of supporting this to be able to configure new Backends too? I mean like the new one by the RISE team? Is this the intended Nexmark use?

Reply:
Yes, new state backends can be added as needed.

Reply:
@iemejia why does Nexmark need the `rocksdb` state backend? That backend is required when state does not fit into the heap, otherwise it is almost always better to use `filesystem`.

Reply:
We may want to experiment with different state backends.